### PR TITLE
ng_nativenet: auto initialization and improvements

### LINKED ISSUE #3093: ng_nativenet interface not auto initialized
`tests/driver_netdev_eth` currently starts the nomac thread for `ng_nativenet` locally. All other devices are initialized automatically.

### PR DESCRIPTION
**Pull request by OlegHahm**
This PR fixes some other minor issues. I can split if you insist.

Fixes #3093

### CODE REVIEW

File: sys/auto_init/netif/auto_init_ng_netdev_eth.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
```
Comment (authmillenon) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: You are @kaspar030?

File: sys/auto_init/netif/auto_init_ng_netdev_eth.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ */
+#define NETDEV_ETH_MAC_STACKSIZE     (THREAD_STACKSIZE_DEFAULT)
+#define NETDEV_ETH_MAC_PRIO          (THREAD_PRIORITY_MAIN - 3)
+
+static char _nomac_stack[NETDEV_ETH_MAC_STACKSIZE];
+
+void auto_init_ng_netdev_eth(void)
+{
+    DEBUG("Initializing NETDEV_ETH device\n");
+    int res = ng_netdev_eth_init(&ng_netdev_eth, (dev_eth_t*)&dev_eth_tap);
```
Comment (authmillenon) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: Shouldn't this hook into `auto_init_dev_eth` somehow?

File: sys/auto_init/netif/auto_init_ng_netdev_eth.c (hunk 1 above)
Comment (OlegHahm) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: No, but it's mostly his code.

File: sys/auto_init/netif/auto_init_ng_netdev_eth.c (hunk 2 above)
Comment (OlegHahm) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: Can you rephrase?

Comment (authmillenon) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: As far as I know `ng_netdev_eth` contains more than just `ng_nativenet` but also other ethernet devices.

Comment (OlegHahm) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: I also thought that, but couldn't find evidence of this.

Comment (kaspar030) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: It does, see #2800.

Comment (OlegHahm) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: So, what does this imply for this PR?

Comment (kaspar030) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: Well, you could adapt ```sys/auto_init/dev_eth/dev_eth_autoinit.c``` and make this PR work not only for dev_eth_tap. But we could postpone that until #2800 is merged...

Comment (OlegHahm) [inline sys/auto_init/netif/auto_init_ng_netdev_eth.c]: I don't not even know how to do this. I completely lost track of netdev_eth, dev_eth, nativenet, ng_nativenet, and so on. Was just happy that this works as intended.

**Comment by authmillenon** [comment]
Please squash

**Comment by OlegHahm** [comment]
squashed

**Comment by OlegHahm** [comment]
I think `defaulttransceiver` may indeed be superfluous. Updated accordingly.

**Comment by authmillenon** [comment]
I'd prefer it with `defaulttransceiver`. You might want to use `ng_netif` without the board's preferred e. g. for unittests or for `ng_slip` or some shiny future module.

**Comment by OlegHahm** [comment]
Agreed, reverted.

**Comment by authmillenon** [comment]
Initialization needs to be removed from test application.

**Comment by OlegHahm** [comment]
The test doesn't use netif auto init. Should we change this?

**Comment by OlegHahm** [comment]
Would be cool if we can get this merged today (or until next Monday).

**Comment by authmillenon** [comment]
Yes, the other tests for netif do. (your change is a success, if you can use `tests/driver_netdev_eth` with a copy of the main.c of `tests/driver_xbee` or `tests/driver_at86rf2xx`.

**Comment by kaspar030** [comment]
this works for me if I select the right USEMODULES. I'd say, squash!

**Comment by kaspar030** [comment]
Is squashing actually needed?

**Comment by OlegHahm** [comment]
No really, squashing, but removing the last two commits.

**Comment by OlegHahm** [comment]
Addressed @authmillenon's comment.

**Comment by kaspar030** [comment]
I'd say, you squash, I do a local compile test and then we merge?

**Comment by authmillenon** [comment]
Last time we did that we introduced errors, so I'd really prefer to wait for Travis, thank you.

**Comment by authmillenon** [comment]
Interface is still not initialized properly in `driver_netdev_eth`

**Comment by authmillenon** [comment]
main.c:68 needs to be removed, too. You are deleting the netif list there

**Comment by OlegHahm** [comment]
updated

**Comment by authmillenon** [comment]
Works. ACK, please squash.

**Comment by OlegHahm** [comment]
Squashed without conflicts.
